File: x264/src/main/java/com/leovp/x264/X264Encoder.java
package com.leovp.x264;

/**
 * Author: Michael Leo
 * Date: 21-3-18 下午4:21
 */
public class X264Encoder {

    public native X264InitResult initEncoder(X264Params params);

    public native void releaseEncoder();

    public native X264EncodeResult encodeFrame(byte[] frame, int colorFormat, long pts);

    public native String getVersion();

    private long ctx;

    static {
        System.loadLibrary("x264a");
    }
}

File: x264/src/main/java/com/leovp/x264/X264Encoder.kt
package com.leovp.x264

/**
 * Author: Michael Leo
 * Date: 21-3-18 下午4:21
 */
class X264Encoder {
    external fun initEncoder(params: X264Params): X264InitResult
    external fun releaseEncoder()
    external fun encodeFrame(frame: ByteArray, colorFormat: Int, pts: Long): X264EncodeResult
    external fun getVersion(): String
    private val ctx: Long = 0

    companion object {
        init {
            System.loadLibrary("libx264-encoder")
            System.loadLibrary("libx264")
        }
    }
}